src/main.rs

```diff
@@ -283,7 +283,10 @@ fn main() {
         let working_dir = tmp_path.join(format!("{}", i));
         let source_root = source_root.clone();
 
-        let t = thread::spawn(move || {
+        let thread_name = String::from("");
+        thread_name.push(i.to_string());
+            
+        let t = thread::::Builder::new().name(thread_name).spawn(move || {
```

Comment:
There's also another thread spawned a few lines earlier to which we should assign a name.